internal/pkg/propagation/summaries.go

```diff
@@ -0,0 +1,761 @@
+	"(*log.Logger).SetOutput": {
+		ifTainted:   0b10,
+		taintedArgs: []int{0},
+	},
+	// func (l *Logger) Writer() io.Writer {
+	"(*log.Logger).Writer": fromFirstArgToFirstRet,
+}
+
+// funcKey represents an interface function by its name and its signature.
+// The signature is a string representation containing only the types of
+// the arguments and return values.
+type funcKey struct {
+	name, signature string
+}
+
+// interfaceFuncSummaries contains summaries for common interface functions
+// such as Write or Read, that could be called statically (i.e. a call to
+// a concrete method whose signature matches an interface method) or dynamically
+// (i.e. a call to an interface method on an interface value).
+var interfaceFuncSummaries = map[funcKey]summary{
+	// type io.Reader interface {
+	//	Read(p []byte) (n int, err error)
+	// }
+	{"Read", "([]byte)(int,error)"}: {
+		ifTainted:   0b1,
+		taintedArgs: []int{1},
+	},
+	// type io.Writer interface {
+	//  Write(p []byte) (n int, err error)
+	// }
+	{"Write", "([]byte)(int,error)"}: {
+		ifTainted:   0b10,
+		taintedArgs: []int{0},
+	},
+	// type io.ReaderFrom interface {
+	// 	ReadFrom(r Reader) (n int64, err error)
+	// }
+	{"ReadFrom", "(Reader)(int64,error)"}: {
+		ifTainted:   0b10,
+		taintedArgs: []int{0},
+	},
+	// type io.WriterTo interface {
+	//  WriteTo(w Writer) (n int64, err error)
+	// }
+	{"WriteTo", "(Writer)(int64,error)"}: {
+		ifTainted:   0b1,
+		taintedArgs: []int{1},
+	},
+	// type io.ReaderAt interface {
+	//	ReadAt(p []byte, off int64) (n int, err error)
+	// }
+	{"ReadAt", "([]byte,int64)(int,error)"}: {
+		ifTainted:   0b1,
+		taintedArgs: []int{1},
+	},
+	// type io.WriterAt interface {
+	//	WriteAt(p []byte, off int64) (n int, err error)
+	// }
+	{"WriteAt", "([]byte,int64)(int,error)"}: {
+		ifTainted:   0b10,
+		taintedArgs: []int{0},
+	},
+	// type io.StringWriter interface {
+	//	WriteString(s string) (n int, err error)
+	// }
+	{"WriteString", "(string)(int,error)"}: {
+		ifTainted:   0b10,
+		taintedArgs: []int{0},
+	},
+	// type fmt.Stringer interface {
+	//	String() string
+	// }
+	{"String", "()(string)"}: {
+		ifTainted:   0b1,
+		taintedRets: []int{0},
+	},
+	// type fmt.GoStringer interface {
+	//	GoString() string
+	// }
+	{"String", "()(string)"}: {
+		ifTainted:   0b1,
+		taintedRets: []int{0},
+	},
+	// type error interface {
+	//	Error() string
+	// }
+	{"Error", "()(string)"}: {
+		ifTainted:   0b1,
+		taintedRets: []int{0},
+	},
+	// Unwrap() error
+	{"Unwrap", "()(error)"}: {
+		ifTainted:   0b1,
+		taintedRets: []int{0},
+	},
+	// Bytes() []byte
+	{"Bytes", "()([]byte)"}: {
+		ifTainted:   0b1,
+		taintedRets: []int{0},
+	},
+	// type context.Context interface {
+	//  Err() error
+	//	Value(key interface{}) interface{}
+	// }
+	{"Err", "()(error)"}: {
+		ifTainted:   0b1,
+		taintedRets: []int{0},
+	},
+	{"Value", "(interface{})(interface{})"}: {
+		ifTainted:   0b1,
+		taintedRets: []int{0},
+	},
+}
```

Comment:
`[Testing]` Review of these summaries was end of day while my eyes twitched from allergies.  I should do another pass for better confidence.